### ISSUE TITLE
feat: upload images on questions

### ISSUE THREAD

**Issue by leonarhv** [drodil/backstage-plugin-qeta#38]
# Why
I know that this might not be trivial, but images are a core thing while asking a question, and really helps with problem resolution. 

# What
I was thinking of creating an annexes button, that saves some image files, to be shown later in the question

**Comment by drodil** [comment]
That's a great idea! Though I think this feature should be configurable from the application; at least there should be options for on/off and for the upload size limit (and maybe for allowed file formats). Saving large attachments might grow the database size pretty quickly. Have to think this through, but it's doable.

**Comment by drodil** [comment]
The markdown editor `react-mde` seems to have some support for file uploads. Should check that out. I think it should work similarly to the editor here in GitHub; file is uploaded and then it's linked as an image (or link if it's not an image) to the markdown content. This would mean at least the following changes:

1. New route to upload attachments, should return uuid
2. New database table to save and serve attachments from
3. New route to download attachments by their uuid
4. Automatic markdown input after attachment is uploaded

Previews etc. should work directly if the attachments are available in public for example in http://localhost:7007/api/qeta/attachments/{uuid} and that returns the attachment using content-disposition and correct content-type headers.

**Comment by drodil** [comment]
Great ideas how this could be implemented! Maybe we could start with saving the files to database and create another issue to support S3 integration through application config? There's already S3 integration for the Backstage catalog so maybe there's something we could re-use for this. Configuration could also support saving files to filesystem or other cloud services later on but still we need a database that contains:

- uuid/id/hash of the image
- location type (db/s3/filesystem/etc.)
- location uri (in case type is not db)
- content (image content when type is db)
- question_id (see below if this makes sense)
- answer_id (see below)
- created timestamp
- creator user reference

Basically we need two new endpoints, one to upload image, return an UUID or similar hash, and another one to download it using that hash. It would be nice to also have a database relation set up between the image and the question/answer but it might be a bit more complicated to implement as the image is uploaded during the creation of the post and it does not have an id at that point. This could help clean the database of images that are not used in case the question/answer is deleted or edited. 

If you are willing to start with the implementation it would be awesome!

**Comment by gabriel-dantas98** [comment]
I made my first functional suggestion https://github.com/drodil/backstage-plugin-qeta/pull/58

Disclaimer: I'm not a frontend developer so some things may not be in the best possible shape. 😅

Currently, it is not very modular for us to plug in other forms of upload, but I can contribute in the following PR`s.

I left a possibility to disable the upload in case someone believes it is not necessary or has some requirement that makes it impossible to use the plugin.

**Comment by drodil** [comment]
This is done and dusted, thanks for the contribution @gabriel-dantas98 !
